### PR TITLE
fix(qa): fix flaky test

### PR DESCRIPTION
**Pull request by miguelpires**
Fixes a flaky test and another bug where the incorrect minimum position was returned when there are multiple exporters configured and one of them has position -1.

### CODE REVIEW

File: broker-core/src/main/java/io/zeebe/broker/exporter/stream/ExporterStreamProcessor.java [hunk 1]
```diff
@@ -107,16 +107,7 @@ public void onOpen(StreamProcessorContext context) {
...
-
-    for (final ExporterContainer container : containers) {
-      container.position = state.getPosition(container.getId());
-
-      if (lowestPosition == -1 || lowestPosition > container.position) {
-        lowestPosition = container.position;
-      }
-    }
-    return lowestPosition;
+    return containers.stream().mapToLong(e -> e.position).min().orElse(-1);
```
Comment (Zelldon) [inline broker-core/src/main/java/io/zeebe/broker/exporter/stream/ExporterStreamProcessor.java]: The position will not contain any value on restart? I think this is the reason why we use the state on the line above

**Review by ChrisKujawa — CHANGES_REQUESTED** [review]
We have some errors on the CI, can you take a look into this?
Furthermore I'm not sure if the `getPositionToRecoveryFrom` changes are correct.

File: broker-core/src/main/java/io/zeebe/broker/exporter/stream/ExporterStreamProcessor.java (hunk 1 above)
Comment (saig0) [reply broker-core/src/main/java/io/zeebe/broker/exporter/stream/ExporterStreamProcessor.java]: What about `containers.stream().mapToLong(c -> state.getPosition(c.getId())).min().orElse(-1)`?

**Review by ChrisKujawa — APPROVED** [review]

**Review by ChrisKujawa — CHANGES_REQUESTED** [review]
Sorry can you apply please phillips suggestion?

**Comment by MiguelPires** [comment]
Sure, all done. Thanks for the feedback @Zelldon and @saig0

**Review by ChrisKujawa — APPROVED** [review]
:+1:

**Comment by MiguelPires** [comment]
bors retry

**Pull request merged**